### ISSUE TITLE
Curve Edit Tools - Insert Spline Knot not being detected in certain segment(s).

### ISSUE THREAD

**Issue by Kickflipkid687** [LeanderSilur/Blender-Tools#9]
I noticed when I was using your addon again in 2.93.  Sometimes when I move the mouse over a certain section of a spline, it doesn't want to insert there... it will go anywhere else, except that part of the spline.  No idea why.  

I reset transforms on the spline, and seems like it's all valid... but I just cannot insert in that spot.   

![image](https://user-images.githubusercontent.com/68572762/133003934-b17eae75-25c5-4d0d-8a7e-810feda7db24.png)

[insert_spline_isssue_2.93.zip](https://github.com/LeanderSilur/Blender-Tools/files/7150353/insert_spline_isssue_2.93.zip)

**Comment by LeanderSilur** [comment]
There was no support for cyclic curves.
Has been added, please try the [release 2.9.2](https://github.com/LeanderSilur/Blender-Tools/releases/tag/curve_edit_tools).

**Comment by LeanderSilur** [comment]
@Kickflipkid687 If you could confirm, that the new release fixes the problem, that would be great.

**Comment by Kickflipkid687** [comment]
Yeah, that worked!  One issue I see yet, which maybe you can't fix/is ok. 

But sometimes when I insert in a new knot/point, some other random points along the spline/curve get their handles sorta mangled. 

So then I had to go through and fixup other areas if i insert a new knot/or yeah, it's a bit random. 

https://user-images.githubusercontent.com/68572762/133907260-c96379b3-c0b6-453a-b65f-23061059e0e1.mp4

**Comment by LeanderSilur** [comment]
Oooh, that looks bad.
Did this problem show up, before the "fix"?
It is the "undo" still working correct in that case?

Another troubleshooting thing: When this happens the next time. Can you save the file, reload it and try again to see the if the problem persists?

**Comment by Kickflipkid687** [comment]
I downloaded your previous operators.py file to compare. 

Yes, I still see the same issue with the previous version/before cyclic update.  
Undo seems to work fine in both cases I think. 
Reloading the file, yes it persists. 

So... I "might"  know what the issue is.... but then again, maybe not, haha.     The curve/file I uploaded on here before.  I have tilts/twists on various spline knots.  Since I wanted to angle my arrayed geometry diff ways along/over the mesh.  

I used that Ctrl + T, to rotate/twist knots in spots.   I re-created another spline object, with similar shape.  If I insert new points, I don't see the issue.  So I'm guessing its from the tilt/twists?  

Then again, i just tried to twist/tilt diff points on this new spline, and it's all good... so I'm not entirely sure, haha...
